src/components/preferences/Webhooks.js

```diff
@@ -0,0 +1,178 @@
+
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+import ids from "./ids";
+import styles from "./styles";
+
+import {
+    Button,
+    Grid,
+    Typography,
+    MenuItem,
+    CircularProgress,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+import ErrorIcon from "@material-ui/icons/Error";
+
+const useStyles = makeStyles(styles);
+
+export default function Webhooks(props) {
+    const { baseId, webhookTopics, webhookTypes, values } = props;
+    const { t } = useTranslation("preferences");
+    const classes = useStyles();
+    const [enableTest, setEnableTest] = React.useState();
+    const [testKey, setTestKey] = React.useState(WEBHOOK_TEST_KEY);
+    const [testError, setTestError] = React.useState();
+    const [testSuccess, setTestSuccess] = React.useState(false);
+
+    const { isFetching: isTesting } = useQuery({
+        queryKey: testKey,
+        queryFn: testWebhook,
+        config: {
+            enabled: enableTest,
+            onSuccess: (resp) => {
+                setEnableTest(false);
+                setTestError(null);
+                setTestSuccess(true);
+            },
+            onError: (e) => {
+                setEnableTest(false);
+                setTestError(e);
+                setTestSuccess(false);
+            },
+        },
+    });
+
+    return (
+        <>
+            <Typography variant="h6" className={classes.sectionHeader}>
+                Webhooks
+            </Typography>
+            <Grid container spacing={3} className={classes.grid}>
+                <Grid item xs={12}>
+                    <Typography>{t("webhooksPrompt")}</Typography>
+                </Grid>
+                <Grid item>
+                    <Field
+                        id={build(baseId, ids.WEBHOOK_URL_TEXT)}
+                        component={FormTextField}
+                        name="webhook.url"
+                    />
+                </Grid>
+                <Grid item>
+                    <Field
+                        name="webhook.type.type"
+                        id={build(baseId, ids.WEBHOOK_TYPES_SELECT)}
+                    >
+                        {({ field: { onChange, ...field }, ...props }) => (
+                            <FormTextField
+                                select
+                                {...props}
+                                label={t("type")}
+                                required
+                                field={field}
+                                onChange={(event) => {
+                                    onChange(event);
+                                }}
+                                variant="outlined"
+                                size="small"
+                            >
+                                {webhookTypes?.map((type, index) => (
```

Comment:
Paul's better at Formik stuff, but I think since you're not altering the behavior of the textfield, I think you can simplify this to:
```suggestion
                    <Field
                        name="webhook.type.type"
                        id={build(baseId, ids.WEBHOOK_TYPES_SELECT)}
                        label={t("type")}
                        required
                        select
                        variant="outlined"
                        size="small"
                        component={FormTextField}
                    >
                      {webhookTypes?.map((type, index) => (
```
I found an example [here](https://github.com/cyverse-de/sonora/blob/c2d5711cd26d5233f51fcfb428c9794fb8a0076b/src/components/apps/editor/params/FileOutputPropertyFields.js#L61)

Reply:
I agree. It's cleaner to set props directly on the `Field`, which will pass them along to its `component` (which is `FormTextField` in this case).